src/test/java/seedu/address/logic/parser/ReserveCommandParserTest.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_FROMDATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ID;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ROOMNUMBER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TODATE;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import java.time.LocalDateTime;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.ReserveCommand;
+import seedu.address.model.hotel.Room;
+import seedu.address.model.hotel.person.Person;
+import seedu.address.model.ids.PersonId;
+import seedu.address.testutil.PersonBuilder;
+import seedu.address.testutil.RoomBuilder;
+
+public class ReserveCommandParserTest {
+    private ReserveCommandParser parser = new ReserveCommandParser();
+    private final Room ROOM = new RoomBuilder().build();
+    private final Person PAYEE = new PersonBuilder().build();
+    private final LocalDateTime FROM_DATE = LocalDateTime.parse("2020-12-12");
+    private final LocalDateTime TO_DATE = LocalDateTime.parse("2020-12-23");
+    private final PersonId ID = PAYEE.getPersonId();
+
+    @Test
+    public void parse_personIdSpecified_success() {
+        String userInput = PREFIX_ID + ID.toString()
+                + PREFIX_ROOMNUMBER + ROOM.getRoomNum()
+                + PREFIX_FROMDATE + FROM_DATE
+                + PREFIX_TODATE + TO_DATE;
+        ReserveCommand expectedCommand = new ReserveCommand(ID, ROOM.getRoomNum(), FROM_DATE, TO_DATE);
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_missingCompulsoryField_failure() {
```

Comment:
See format for reserveCommand, I guess this is just copy and haven't done yet?

Reply:
I don't understand what you're saying. The test follows the ReserveCommand format.

Reply:
sorry it is indeed correct